### ISSUE TITLE
[Enhancement] New rule: --force-assertion-description

### ISSUE THREAD

**Issue by bootuz** [nunnatsa/ginkgolinter#199]
In our team the optional description in gomega matchers are required and a PR won't be merged until it's fixed. And devs often forget to add them.

So, I'd like to an optional rule that will force devs to add description to gomega matchers.

For example, in the linter settings, if we add this rule: `force-assertion-description`:

```go
var _ = Describe("optional description tests", func() {
	It("should test missing descriptions", func() {
		// This should trigger warning when ForceOptionalDescription is enabled
		Expect("hello").To(Equal("hello")) // want `missing optional description`
		
		// This should trigger warning - no description
		Expect(42).Should(Equal(42)) // want `missing optional description`
		
		// This should trigger warning - no description  
		Expect([]int{1, 2, 3}).To(HaveLen(3)) // want `missing optional description`
		
		// This should NOT trigger warning - has description
		Expect(true).To(BeTrue(), "this should be true")
	})
})
```

**Comment by bootuz** [comment]
PR https://github.com/nunnatsa/ginkgolinter/pull/200
